src/components.d.ts

```diff
@@ -19,6 +19,10 @@ export namespace Components {
           * Sets the button to a full-width block. If the attribute’s value is undefined, the value is set to false.
          */
         "expand": boolean;
+        /**
+          * Accepts the id of the fw-modal component to open it on click
+         */
+        "modalTriggerId": string;
```

Comment:
Can we also expose `openModal` method for the cases where a user wants to open up a modal without a button in the picture

Reply:
It can be done through the visible property in the modal. Setting visible to true will show it.